### PR TITLE
feat: Storage watcher bot

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: storage-watcher/src/agent-storage-watcher.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      await Promise.all(
+        contractStorageMap.slots.map(async (slot: StorageSlot) => {
+          const value = await getStorageValue(
+            contractStorageMap.contract.address,
+            slot,
+            blockNumber
+          );
+          if (checkValues) {
+            const prevValue = contractInfo.get(slot.name);
+            if (value != prevValue && prevValue) {
```
Comment (madlabman) [inline storage-watcher/src/agent-storage-watcher.ts:56]: Why inverted?

File: storage-watcher/src/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export async function getStorageValue(
+  address: string,
+  slot: StorageSlot,
+  block?: number
+) {
+  const slotAddress = slot.address
+    ? slot.address
+    : keccak256(toUtf8Bytes(slot.name));
+  return await ethersProvider.getStorageAt(
```
Comment (madlabman) [inline storage-watcher/src/helpers.ts:14]: Which value will be returned if no slot value is set? I'm asking regarding `!prevValue` check.

File: storage-watcher/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    /* Experimental Options */
+    "experimentalDecorators": true,              /* Enables experimental support for ES7 decorators. */
+    "emitDecoratorMetadata": true,               /* Enables experimental support for emitting type metadata for decorators. */
+
+    /* Advanced Options */
+    "skipLibCheck": true,                           /* Skip type checking of declaration files. */
+    "forceConsistentCasingInFileNames": true,        /* Disallow inconsistently-cased references to the same file. */
+    "resolveJsonModule": true
+  }
+}
```
Comment (madlabman) [inline storage-watcher/tsconfig.json:73]: Let's clean up the thing.

File: storage-watcher/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+LABEL "network.forta.settings.agent-logs.enable"="true"
+ENV NODE_ENV=production
+WORKDIR /app
+COPY package*.json .yarnrc.yml yarn.lock ./
+COPY .yarn/releases ./.yarn/releases/
+COPY --from=builder /app/.yarn/cache ./.yarn/cache
+RUN yarn install --immutable && yarn cache clean
+COPY --from=builder /app/dist ./src
+COPY version.json ./
+CMD [ "yarn", "run", "start:prod" ]
```
[the remark this answers is not in the document]
Comment (dgusakov) [reply storage-watcher/Dockerfile:21]: Copy pasta from the template bot

File: storage-watcher/src/helpers.ts (hunk 2 above)
Comment (dgusakov) [reply storage-watcher/src/helpers.ts:14]: 0x00000000...000

File: storage-watcher/tsconfig.json (hunk 3 above)
Comment (dgusakov) [reply storage-watcher/tsconfig.json:19]: Done

File: storage-watcher/src/agent-storage-watcher.ts (hunk 1 above)
Comment (dgusakov) [reply storage-watcher/src/agent-storage-watcher.ts]: Fixed

**Review by madlabman — APPROVED** [review]

**Pull request merged**